tests/unit_tests/cuda/rk_stepper_cuda.cpp

```diff
@@ -60,23 +62,40 @@ TEST(rk_stepper_cuda, rk_stepper) {
     for (unsigned int i = 0; i < theta_steps * phi_steps; i++) {
 
         auto& traj = tracks_host[i];
+        free_track_parameters c_traj(traj);
 
         // Forward direction
-        rk_stepper_type::state forward_state(traj);
+        rk_stepper_t::state rk_state(traj);
+        crk_stepper_t::state crk_state(c_traj);
+
+        crk_state.template set_constraint<step::constraint::e_user>(
+            0.5 * unit_constants::mm);
+        n_state._step_size = 1. * unit_constants::mm;
+        ASSERT_NEAR(crk_state.constraints().template size<>(),
+                    0.5 * unit_constants::mm, epsilon);
         for (unsigned int i_s = 0; i_s < rk_steps; i_s++) {
-            rk.step(forward_state, n_state);
+            rk_stepper.step(rk_state, n_state);
+            crk_stepper.step(crk_state, n_state);
+            crk_stepper.step(crk_state, n_state);
```

Comment:
Just a question: why the stepper with constraint step size needs two step calls?

Reply:
Because it is restrained to half the step size of the other stepper, so it needs to make two steps to reach the same position as the unconstrained stepper. This might not work entirely, because the rk stepper might adjust the step size further, but apparently with the given epsilon in this B-field it is good enough

Reply:
I see. There is 1 mm default stepsize in dummy navigation state..